core/src/main/scripts/importer/validateData.py

```diff
@@ -1228,6 +1232,409 @@ def checkLine(self, data):
+                                  "and/or Tumor_Seq_Allele2 contain deletion (-)." % variant_type
+                    if not self.strict_mode:
+                        self.logger.warning(log_message, extra={'line_number': self.line_number,
+                                                                'cause': '(%s, %s, %s)' % (ref_allele, 
+                                                                                           tumor_seq_allele1, 
+                                                                                           tumor_seq_allele2)})
+                    else:
+                        self.logger.error(log_message, extra={'line_number': self.line_number,
+                                                              'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                         tumor_seq_allele2)})
+        return True
+
+    def checkAlleleSpecialCases(self, data):
+        """ Check other special cases which should or should not occur in Allele Based columns
+        Special cases are either from unofficial vcf2maf rules or discrepancies identified. """
+
+        # First check if columns necessary exist in the data
+        necessary_columns = ['Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Variant_Type']
+        if set(necessary_columns).issubset(self.cols):
+            ref_allele = data[self.cols.index('Reference_Allele')].strip()
+            tumor_seq_allele1 = data[self.cols.index('Tumor_Seq_Allele1')].strip()
+            tumor_seq_allele2 = data[self.cols.index('Tumor_Seq_Allele2')].strip()
+            variant_type = data[self.cols.index('Variant_Type')].strip()
+
+            # Check if Allele Based columns are not all the same
+            if ref_allele == tumor_seq_allele1 and tumor_seq_allele1 == tumor_seq_allele2:
+                log_message = "All Values in columns Reference_Allele, Tumor_Seq_Allele1 " \
+                              "and Tumor_Seq_Allele2 are equal."
+                if not self.strict_mode:
+                    self.logger.warning(log_message, extra={'line_number': self.line_number,
+                                                            'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1,
+                                                                                       tumor_seq_allele2)})
+                else:
+                    self.logger.error(log_message, extra={'line_number': self.line_number,
+                                                          'cause': '(%s, %s, %s)' % (ref_allele, tumor_seq_allele1, 
+                                                                                     tumor_seq_allele2)})
+
+            # Check in case of insertion if the Reference_Allele contains '-'
+            if variant_type == "INS" and '-' not in ref_allele:
```

Comment:
I'm not sure if this test is correct. Does_ the reference always have to be `-` in case of an insertion? What about A -> GT ?

Reply:
You indeed may be correct: 
A -> GT will be checked correct by the following check: 
`If Variant_Type == "INS", then (End_position - Start_position + 1 == length (Reference_Allele) or End_position - Start_position == 1) and length(Reference_Allele) <= length(Tumor_Seq_Allele1 and Tumor_Seq_Allele2)`

And the MAF file information says that the flanking reference allele cannot be in there, but says nothing about when it is a different one.
<img width="500" alt="screen shot 2018-03-13 at 11 15 04 am" src="https://user-images.githubusercontent.com/1343609/37335857-df11c796-26af-11e8-8de9-9899cecb83d8.png">